File: api/src/main/java/com/caibowen/prma/api/LogLevel.java
package com.caibowen.prma.api;

/**
 * @author BowenCai
 * @since 26-10-2014.
 */
public enum LogLevel {

    /**
     * The <code>ERROR</code> level designates error events which may or not
     * be fatal from the application.
     */
    ERROR(8, "ERROR"),

    /**
     * The <code>WARN</code> level designates potentially harmful situations.
     */
    WARN(6, "WARN"),

    /**
     * The <code>INFO</code> level designates informational messages
     * highlighting overall progress of the application.
     */
    INFO(4, "INFO"),

    /**
     * The <code>DEBUG</code> level designates informational events of lower
     * importance.
     */
    DEBUG(2, "DEBUG"),

    /**
     * The <code>TRACE</code> level designates informational events of very low
     * importance.
     */
    TRACE(1, "TRACE");

    public final int levelInt;
    public final String levelStr;
    private LogLevel(int i, String s) {
        this.levelInt = i;
        this.levelStr = s;
    }

    public static LogLevel from(String name) {
        return valueOf(name.toUpperCase());
    }

    public static LogLevel from(int iv) {
        return values()[4 - iv / 2];
    }

}

File: api/src/main/scala/com/caibowen/prma/api/LogLevel.scala
package com.caibowen.prma.api

/**
 * @author BowenCai
 * @since  02/12/2014.
 */
object LogLevel extends Enumeration {
  type LogLevel = Value
  val OFF = Value(32, "OFF")
  val FATAL = Value(16, "FATAL")
  /* MAIN */
  val ERROR = Value(8, "ERROR")
  val WARN = Value(6, "WARN")
  val INFO = Value(4, "INFO")
  val DEBUG = Value(2, "DEBUG")
  val TRACE = Value(1, "TRACE")
  /* MAIN */
  val ALL = Value(0, "ALL")

  def from(name: Int): LogLevel = {
    super.apply(name)
  }

  def from(name: String): LogLevel = {
    val tname = name.trim.toUpperCase
    super.withName(tname)
  }
}
